crs-setup.conf.example

```diff
@@ -752,6 +752,52 @@ SecDefaultAction "phase:2,log,auditlog,pass"
 #  setvar:'tx.dos_block_timeout=600'"
 
 
+#
+# -- [[ Anti-virus Protection ]] ------------------------------------
+#
+# Optional anti-virus protection for uploaded files.
+#
+# Currently, only ClamAV anti-virus is supported.
+#
+# Prerequisities for this feature:
+# - Lua support must be enabled
+# - Library lua-socket must be installed in the system
+# - modsecurity 'SecTmpSaveUploadedFiles' directive is 'On' or
+#   the 'SecUploadKeepFiles' directive is set to 'RelevantOnly'
+#
+# With setting 'tx.crs_anti-virus_clamav_connect_type' you can choose
+# between two connection types:
+# - tcp: connect using TCP/IP protocol
+# - socket: connect using unix socket file
+#
+# Make sure that setting 'tx.crs_anti-virus_clamav_chunk_size' below
+# does not exceed 'StreamMaxLength' as defined in ClamAV
+# configuration file clamd.conf.
+#
+# After enabling, anti-virus scanning should be tested, for example,
+# using:
+# curl http://localhost --form "data=@eicar.com"
+# Get eicar test file from https://www.eicar.org.
+#
+# Default: no
+# Uncomment this rule to change the default.
+#
+#SecAction \
+# "id:900800,\
+#  phase:1,\
+#  nolog,\
+#  pass,\
+#  t:none,\
+#  setvar:'tx.crs_anti-virus_protection_enable=1',\
+#  setvar:'tx.crs_anti-virus_clamav_connect_type=socket',\
+#  setvar:'tx.crs_anti-virus_clamav_socket_file=/var/run/clamav/clamd.ctl',\
+#  setvar:'tx.crs_anti-virus_clamav_address=127.0.0.1',\
+#  setvar:'tx.crs_anti-virus_clamav_port=3310',\
+#  setvar:'tx.crs_anti-virus_clamav_chunk_size=4096',\
+#  setvar:'tx.crs_anti-virus_clamav_network_timeout_seconds=2',\
+#  setvar:'tx.crs_anti-virus_clamav_max_file_size_bytes=1048576'"
```

Comment:
To avoid the FP's in case of libmodsecurity3 (because the `REQUEST_BODY` is always presented), we should add these actions:
```
  ctl:ruleRemoveById=920272,\
  ctl:ruleRemoveById=920273"
```
or we can combine these exceptions with checking of CT header.

Reply:
I would not count on CT as it can be faked to bypass anti-virus check.

Reply:
May be we should discuss this on monthly chat. I tried this rule in crs-setup.conf:
```
SecRule REQUEST_HEADERS:Content-Type "(?:multipart/form-data)" \
    "id:'900801',\
    phase:1,\
    t:none,t:lowercase,\
    pass,\
    nolog,\
    ctl:ruleRemoveTargetById=920272;REQUEST_BODY,\
    ctl:ruleRemoveTargetById=920273;REQUEST_BODY"
```

Reply:
I will keep this open, i think we need to discuss it with others.